shop/src/main/java/ro/msg/learning/shop/service/CustomerService.java

```diff
@@ -0,0 +1,38 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.dto.save.SaveCustomerDto;
+import ro.msg.learning.shop.model.Customer;
+import ro.msg.learning.shop.repository.CustomerRepository;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CustomerService {
+    private final CustomerRepository customers;
+
+    public CustomerService(CustomerRepository customers) {
+        this.customers = customers;
+    }
+
+    public Customer saveCustomer(SaveCustomerDto customer) {
+        return customers.save(customer.toCustomer());

```

Comment:
Why do you need a separate DTO for customer? 
Also, the mapping from entity to DTO and vice-versa should be done in the Mapper. Create a mapper class that takes this logic.

Reply:
Note: all of the entities that you use will require mappings to and from their DTOs